crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -1945,4 +1927,43 @@ mod tests {
 
         Ok(())
     }
+
+    #[async_test]
+    async fn test_lock_multiple_requests() -> Result<()> {
+        let server = MockServer::start().await;
+        let client = logged_in_client(Some(server.uri())).await;
+
+        let pos = Arc::new(Mutex::new(0));
+        let _mock_guard = Mock::given(SlidingSyncMatcher)
+            .respond_with(move |_: &Request| {
+                let mut pos = pos.lock().unwrap();
+                *pos += 1;
+                ResponseTemplate::new(200).set_body_json(json!({
+                    "pos": pos.to_string(),
+                    "lists": {},
+                    "rooms": {}
+                }))
+            })
+            .mount_as_scoped(&server)
+            .await;
+
+        let sliding_sync = client
+            .sliding_sync("test")?
+            .with_to_device_extension(
+                assign!(v4::ToDeviceConfig::default(), { enabled: Some(true)}),
+            )
+            .with_e2ee_extension(assign!(v4::E2EEConfig::default(), { enabled: Some(true)}))
+            .build()
+            .await?;
+
+        // Spawn two requests in parallel. Before #2430, this lead to a deadlock and the
```

Comment:
```suggestion
        // Spawn two requests in parallel. Before #2430, this leads to a deadlock and the
```

Reply:
I meant this as the past form for "lead" (which is "lead"), so I'll leave it to that :p